src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -0,0 +1,560 @@
+
+            // constexpr double MinAutoScrollDist = 2.0; // Arbitrary value
+            // double newAutoScrollVelocity = 0.0;
+            // if (cursorBelowBottomDist > MinAutoScrollDist)
+            // {
+            //     newAutoScrollVelocity = _getAutoScrollSpeed(cursorBelowBottomDist);
+            // }
+            // else if (cursorAboveTopDist > MinAutoScrollDist)
+            // {
+            //     newAutoScrollVelocity = -1.0 * _getAutoScrollSpeed(cursorAboveTopDist);
+            // }
+
+            // if (newAutoScrollVelocity != 0)
+            // {
+            //     _tryStartAutoScroll(point, newAutoScrollVelocity);
+            // }
+            // else
+            // {
+            //     _tryStopAutoScroll(ptr.PointerId());
+            // }
+        }
+
+        _core->UpdateHoveredCell(terminalPosition);
+    }
+
+    void ControlInteractivity::TouchMoved(const winrt::Windows::Foundation::Point newTouchPoint,
+                                          const bool focused)
+    {
+        if (focused &&
+            _touchAnchor)
+        {
+            const auto anchor = _touchAnchor.value();
+
+            // Our actualFont's size is in pixels, convert to DIPs, which the
+            // rest of the Points here are in.
+            const auto fontSizeInDips{ _core->FontSizeInDips() };
+
+            // Get the difference between the point we've dragged to and the start of the touch.
+            const float dy = newTouchPoint.Y - anchor.Y;
+
+            // Start viewport scroll after we've moved more than a half row of text
+            if (std::abs(dy) > (fontSizeInDips.height<float>() / 2.0f))
+            {
+                // Multiply by -1, because moving the touch point down will
+                // create a positive delta, but we want the viewport to move up,
+                // so we'll need a negative scroll amount (and the inverse for
+                // panning down)
+                const float numRows = -1.0f * (dy / fontSizeInDips.height<float>());
+
+                const auto currentOffset = ::base::ClampedNumeric<double>(_core->ScrollOffset());
+                const auto newValue = numRows + currentOffset;
+
+                // Update the Core's viewport position, and raise a
+                // ScrollPositionChanged event to update the scrollbar
+                _updateScrollbar(newValue);
+
+                // Use this point as our new scroll anchor.
+                _touchAnchor = newTouchPoint;
+            }
+        }
+    }
+
+    void ControlInteractivity::PointerReleased(TerminalInput::MouseButtonState buttonState,
+                                               const unsigned int pointerUpdateKind,
+                                               const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                                               const bool /*focused*/,
+                                               const til::point terminalPosition)
+    {

```

Comment:
null-check `_core`?

Reply:
We instantiate `_core` in the ctor, so this isn't really needed